gnucash/report/report-system/test/test-report-jqplot.scm

```diff
@@ -0,0 +1,699 @@
+    (gnc:html-barchart-append-row! html-bar-chart '(6.00))
+    (gnc:html-barchart-prepend-row! html-bar-chart '(4.00))
+    (gnc:html-barchart-append-column! html-bar-chart '(7.00 8.00 9.00))
+    (gnc:html-barchart-prepend-column! html-bar-chart '(1.00 2.00 3.00))
+    ;; check test data
+    (test-equal "HTML Bar Chart - data row 0"
+      '(1.0 4.0 7.0)
+      (car (gnc:html-barchart-data html-bar-chart)))
+    (test-equal "HTML Bar Chart - data row 1"
+      '(2.0 5.0 8.0)
+      (cadr (gnc:html-barchart-data html-bar-chart)))
+    (test-equal "HTML Bar Chart - data row 2"
+      '(3.0 6.0 9.0)
+      (caddr (gnc:html-barchart-data html-bar-chart)))
+  )
+
+  (test-end "Jqplot - HTML Bar Chart Data")
+)
+
+;; -----------------------------------------------------------------------
+
+(define (test-line-chart-data)
+
+  (test-begin "Jqplot - HTML Line Chart Data")
+
+  (let (
+         (html-line-chart (gnc:make-html-linechart))
+       )
+    ;; prepare test data
+    (gnc:html-linechart-set-data! html-line-chart '((5.00)))
+    (gnc:html-linechart-append-row! html-line-chart '(6.00))
+    (gnc:html-linechart-prepend-row! html-line-chart '(4.00))
+    (gnc:html-linechart-append-column! html-line-chart '(7.00 8.00 9.00))
+    (gnc:html-linechart-prepend-column! html-line-chart '(1.00 2.00 3.00))
+    ;; check test data
+    (test-equal "HTML Line Chart - data row 0"
+      '(1.0 4.0 7.0)
+      (car (gnc:html-linechart-data html-line-chart)))
+    (test-equal "HTML Line Chart - data row 1"
+      '(2.0 5.0 8.0)
+      (cadr (gnc:html-linechart-data html-line-chart)))
+    (test-equal "HTML Line Chart - data row 2"
+      '(3.0 6.0 9.0)
+      (caddr (gnc:html-linechart-data html-line-chart)))
+  )
+
+  (test-begin "Jqplot - HTML Line Chart Data")
+)
+
+;; -----------------------------------------------------------------------
+
+(define (test-bar-chart-rendering)
+
+  (test-begin "Jqplot - HTML Bar Chart Rendering")
+
+  (let (
+         (html-bar-chart (gnc:make-html-barchart))
+       )
+    ;; prepare test data
+    (gnc:html-barchart-set-width! html-bar-chart (cons 'pixels 800))
+    (gnc:html-barchart-set-height! html-bar-chart (cons 'pixels 400))
+    (gnc:html-barchart-set-title! html-bar-chart "Test bar Chart")
+    (gnc:html-barchart-set-subtitle! html-bar-chart "Created 2018-08-02")
+    (gnc:html-barchart-set-x-axis-label! html-bar-chart "Test x-Axis Label barchart")
+    (gnc:html-barchart-set-y-axis-label! html-bar-chart "Test y-Axis Label barchart")
+    (gnc:html-barchart-set-col-labels! html-bar-chart '("Col1" "Col2" "Col3"))
+    (gnc:html-barchart-set-row-labels! html-bar-chart '("Row1" "Row2" "Row3"))
+    (gnc:html-barchart-set-data! html-bar-chart '((5.00)))
+
+    ;; do the test
+    (test-equal "Jqplot - Render Bar Chart"
+      (string-concatenate (drop (gnc:html-document-tree-collapse (gnc:html-barchart-render html-bar-chart #f)) 11))
+"\" style=\"width:800px;height:400px;\"></div>\n\
```

Comment:
This is terribly brittle and any errors will be difficult to diagnose. That's partly due to gnc:html-foo-render doing too much at once, which makes it impossible to test the different points in its execution. Since Chris Lam is working on replacing these functions (and indeed replacing jqplot) entirely it's not worth the effort to refactor them into something more testable. I suggest that you just remove the tests.

Reply:
Agree.
I was putting it here as a regression test that is only supposed to act as indicator for "something has changed". The idea was: To have an indication is better than not having any indication. For diagnosis this will not help, but at least you know that there might be something to analyze.
I do not have a strong opinion here. But to keep the possibility for easy regression, I would suggest to only comment this test out, so there is a quick possibility to enable it again "offline" if deemed useful for whatever reason. I can also decorate it with corresponding comments so other contributors/developers are aware of that idea.
Is that ok?

Reply:
It's best to remove them completely. It's just clutter and will really annoy anyone trying to develop charts.

Reply:
ah, come on,  let's close the issue